### PR TITLE
Ensure versioncmp 'a' parameter is a string

### PR DESCRIPTION
**Pull request by raoulbhatia**
Fixes:
> Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation Error: Error while evaluating a Function Call, 'versioncmp' parameter 'a' expects a String value, got Undef (file: /etc/puppetlabs/code/environments/production/modules/augeas/manifests/lens.pp, line: 61, column: 24) (file: /etc/puppetlabs/code/environments/production/modules/postfix/manifests/augeas.pp, line: 7) on node [...]

### CODE REVIEW

**Comment by raphink** [comment]
Well that might fix it, but your error is not supposed to happen (i.e. `$::augeasversion` being undef).

**Comment by raoulbhatia** [comment]
I am still hitting this error with Ubuntu 18.04 and Puppet 5.4.0-2ubuntu3.  Any way that I can better debug this?

**Comment by raoulbhatia** [comment]
Ok, it seems that `$::augeasversion` is undef if `augeas-tools` is not installed.

**Comment by raphink** [comment]
It should not require `augeas-tools`, only the Augeas library.

**Comment by kjetilho** [comment]
The problem is with Facter 3.x, the C implementation runs "augparse --version" to find the version instead of using the Ruby library.  The old Ruby implementation works just fine without augeas-tools installed.  One possible workaround is to include that implementation as a plugin fact...

Tested on Ubuntu Bionic with facter 3.10.0-4

**Comment by linuxmail** [comment]
hi,

I've changed the line https://github.com/camptocamp/puppet-augeas/blob/8e605057c0b3f45b4cf36b13e1ad292fc3b63b88/manifests/lens.pp#L61

from:
```
if (!$stock_since or versioncmp($::augeasversion, $stock_since) < 0) {
```
to:

```
if (!$stock_since or versioncmp("${::augeasversion}", $stock_since) < 0) {
```
than the problem disapears.

cu denny

**Comment by oasys** [comment]
I"m seeing this same issue on a Debian Buster install.

Presumably this isn't merged because of the failing ci test:

>manifests/lens.pp - WARNING: string containing only a variable on line 61

Is there something that can be done to fix the test or is there a different/better way to solve the problem?

File: manifests/lens.pp [hunk 1]
```diff
@@ -58,7 +58,7 @@
     mode => '0644',
   }
 
-  if (!$stock_since or versioncmp($::augeasversion, $stock_since) < 0) {
+  if (!$stock_since or versioncmp("${::augeasversion}", $stock_since) < 0) {
```
Comment (raphink) [inline manifests/lens.pp:61]:
```suggestion
  if (!$stock_since or versioncmp(String($::augeasversion), $stock_since) < 0) {
```

**Comment by raoulbhatia** [comment]
Thanks for improving my PR! 🇨🇭

**Comment by oasys** [comment]
Thanks, this works for me.  Can we do a forge release?

**Pull request merged**